datadog_checks_base/datadog_checks/base/utils/db/query.py

```diff
@@ -44,6 +44,8 @@ def compile(self, column_transformers, extra_transformers):
         query = self.query_data.get('query')
         if not query:
             raise ValueError('field `query` for {} is required'.format(query_name))
+        elif callable(query):
+            pass  # OK, query comes from code, assume they know what they're doing.
```

Comment:
I'll submit a separate PR for this. Note to self that this means I'll also have to bump `datadog_checks_base` here.